rotkehlchen/chain/aggregator.py

```diff
@@ -1339,19 +1359,57 @@ def add_accounts_to_all_evm(
 
         Returns a list of tuples of the address and the chain it was added in.
         """
-        filtered_accounts = self.filter_active_evm_addresses(accounts)
-        added_accounts = []
-        for chain, account in filtered_accounts:
-            try:
-                self.modify_blockchain_accounts(
-                    blockchain=chain,
-                    accounts=[account],
-                    append_or_remove='append',
+        added_accounts: list[tuple[SUPPORTED_EVM_CHAINS, ChecksumEvmAddress]] = []
+        # Distinguish between contracts and EOAs
+        for account in accounts:
+            if self.is_contract(account, SupportedBlockchain.ETHEREUM):
+                added_chains = self.track_evm_address(account, [SupportedBlockchain.ETHEREUM])
+                if len(added_chains) == 1:  # Is always either 1 or 0 since is only for ethereum
+                    added_accounts.append((SupportedBlockchain.ETHEREUM, account))
+            else:
+                added_accounts += self.check_chains_and_add_accounts(
+                    account=account,
+                    chains=list(typing.get_args(SUPPORTED_EVM_CHAINS)),
```

Comment:
say that I already have the account in ethereum and optimism... would it try to check them again? I understand that it is what will happen. Maybe you can exclude the chains where this address has been already added?

Reply:
Makes sense, changed that